test/e2e/storage/drivers/in_tree.go

```diff
@@ -0,0 +1,1388 @@
+						{Name: "mountd", ContainerPort: 20048},
+						{Name: "rpcbind", ContainerPort: 111},
+						{Name: "rpcbind-udp", ContainerPort: 111, Protocol: v1.ProtocolUDP},
+					},
+					Env: []v1.EnvVar{
+						{
+							Name: "POD_IP",
+							ValueFrom: &v1.EnvVarSource{
+								FieldRef: &v1.ObjectFieldSelector{
+									FieldPath: "status.podIP",
+								},
+							},
+						},
+					},
+					ImagePullPolicy: v1.PullIfNotPresent,
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      "export-volume",
+							MountPath: "/export",
+						},
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: "export-volume",
+					VolumeSource: v1.VolumeSource{
+						EmptyDir: &v1.EmptyDirVolumeSource{},
+					},
+				},
+			},
+		},
+	}
+	provisionerPod, err := podClient.Create(provisionerPod)
+	framework.ExpectNoError(err, "Failed to create %s pod: %v", provisionerPod.Name, err)
+
+	framework.ExpectNoError(framework.WaitForPodRunningInNamespace(c, provisionerPod))
+
+	By("locating the provisioner pod")
+	pod, err := podClient.Get(provisionerPod.Name, metav1.GetOptions{})
+	framework.ExpectNoError(err, "Cannot locate the provisioner pod %v: %v", provisionerPod.Name, err)
+
+	return pod
+}
+
+func (n *nfsDriver) CleanupDriver(vt testpatterns.TestVolType) {
+	f := n.driverInfo.Framework
+	cs := f.ClientSet
+	ns := f.Namespace
+
+	switch vt {
+	case testpatterns.InlineVolume:
+		fallthrough
+	case testpatterns.PreprovisionedPV:
+		framework.CleanUpVolumeServer(f, n.serverPod)
+	case testpatterns.DynamicPV:
+		framework.DeletePodOrFail(cs, ns.Name, n.externalProvisionerPod.Name)
+		clusterRoleBindingName := ns.Name + "--" + "cluster-admin"
+		cs.RbacV1beta1().ClusterRoleBindings().Delete(clusterRoleBindingName, metav1.NewDeleteOptions(0))
+	default:
+		framework.Failf("Unsupported volType:%v is specified", vt)
+	}
+}
+
+func (n *nfsDriver) CreateVolume(vt testpatterns.TestVolType) {
+	// Volume is created inside the pod that is created in NewNFSServer(), so nothing to do here.
+}
+
+func (n *nfsDriver) DeleteVolume(vt testpatterns.TestVolType) {
+	// Volume inside the pod is deleted in CleanUpVolumeServer(), so nothing to do here.
+}
+
+// Gluster
+type glusterFSDriver struct {
+	serverIP  string
+	serverPod *v1.Pod
+
+	driverInfo DriverInfo
+}
+
+// InitGlusterFSDriver returns glusterFSDriver that implements TestDriver interface
+func InitGlusterFSDriver() TestDriver {
+	return &glusterFSDriver{
+		driverInfo: DriverInfo{
+			Name:        "gluster",
+			MaxFileSize: testpatterns.FileSizeMedium,
+			SupportedFsType: map[string]bool{
+				"": true, // Default fsType
+			},
+			IsPersistent:       true,
+			IsFsGroupSupported: false,
+			IsBlockSupported:   false,
+		},
+	}
+}
+
+func (g *glusterFSDriver) GetDriverInfo() *DriverInfo {
+	return &g.driverInfo
+}
+
+func (g *glusterFSDriver) SkipUnsupportedTest(tp testpatterns.TestPattern) {
+	framework.SkipUnlessNodeOSDistroIs("gci", "ubuntu", "custom")
```

Comment:
I think this first skip can be removed.  All gce/gke suites should now be only those 3.

Reply:
I think that original intention of this skip was to skip gluster test on a OS distro that doesn't support glusterfs. For debian or fedora base OS distro, it would be needed. And it won't be cloud provider independent factor.
(They don't have a CI system, so removing this wouldn't impact any CI, though.)

https://github.com/kubernetes/kubernetes/commit/ecca32641f9dd3143da94b08eeb7f3d33fb458a3#diff-c83ebba11ffd12bd594ac8aa2f9301ffR143